examples/c/5_series/gnss_ins/5_series_gnss_ins_example.c

```diff
@@ -1,354 +1,1106 @@
+        2000,                                 // Set the base timeout for command replies (milliseconds)
+        &mip_interface_user_send_to_device,   // User-defined send packet callback
+        &mip_interface_user_recv_from_device, // User-defined receive packet callback
+        &mip_interface_default_update,        // Default update callback
+        (void*)_device_port                   // Cast the device port for use in the callbacks
+    );
+
+    // Ping the device
+    // Note: This is a good first step to make sure the device is present
+    MICROSTRAIN_LOG_INFO("Pinging the device.\n");
+    mip_cmd_result cmd_result = mip_base_ping(_device);
+
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(_device, cmd_result, "Could not ping the device!\n");
+    }
+
+    // Set the device to Idle
+    // Note: This is good to do during setup as high data traffic can cause commands to fail
+    MICROSTRAIN_LOG_INFO("Setting the device to idle.\n");
+    cmd_result = mip_base_set_idle(_device);
+
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(_device, cmd_result, "Could not set the device to idle!\n");
+    }
+
+    // Print device info to make sure the correct device is being used
+    MICROSTRAIN_LOG_INFO("Getting the device information.\n");
+    mip_base_device_info device_info;
+    cmd_result = mip_base_get_device_info(_device, &device_info);
+
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(_device, cmd_result, "Could not get the device information!\n");
+    }
+
+    // Extract the major minor and patch values
+    const uint16_t major = device_info.firmware_version / 1000;
+    const uint16_t minor = device_info.firmware_version / 100 % 10;
+    const uint16_t patch = device_info.firmware_version % 100;
+
+    // Firmware version format is x.x.xx
+    char firmwareVersion[16];
+    snprintf(firmwareVersion, sizeof(firmwareVersion) / sizeof(firmwareVersion[0]), "%d.%d.%02d",
+        major,
+        minor,
+        patch
+    );
+
+    MICROSTRAIN_LOG_INFO("-------- Device Information --------\n");
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Name", device_info.model_name);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Model Number", device_info.model_number);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Serial Number", device_info.serial_number);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Lot Number", device_info.lot_number);
+    MICROSTRAIN_LOG_INFO("%-16s | %.16s\n", "Options", device_info.device_options);
+    MICROSTRAIN_LOG_INFO("%-16s | %16s\n", "Firmware Version", firmwareVersion);
+    MICROSTRAIN_LOG_INFO("------------------------------------\n");
+
+    // Load the default settings on the device
+    // Note: This guarantees the device is in a known state
+    MICROSTRAIN_LOG_INFO("Loading default settings.\n");
+    cmd_result = mip_3dm_default_device_settings(_device);
```

Comment:
This forces the UART baudrates to 115200, and will cause an error if using serial at any other rate (e.g. 921600). I feel like this is a good thing to do, but maybe it should handle the baud rate switch situation somehow.

Reply:
This needs to be documented in the device docs then. This makes no sense to essentially break the connection to load defaults over serial